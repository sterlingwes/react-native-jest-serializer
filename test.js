import React from 'react';
import { View } from 'react-native';
import renderer from 'react-test-renderer';

expect.addSnapshotSerializer(require('./serializer'));

describe('style prop serializing', () => {
  describe('style with overrides and nested objects', () => {
    const Component = () => (
      <View
        style={[
          0,
          { height: 1, position: 'absolute' },
          undefined,
          { flex: 1, height: 10 },
          null,
          false,
          10,
          {
            transform: [{ perspective: 850 }, { translateX: -100 }, { rotateY: '60deg' }],
          },
          true,
        ]}
      ></View>
    );

    const tree = renderer.create(<Component />).toJSON();

    it('merge an array to the relevant style object', () => {
      expect(tree).toMatchInlineSnapshot(`
                        <View
                          style={
                            Object {
                              "flex": 1,
                              "height": 10,
                              "position": "absolute",
                              "transform": Array [
                                Object {
                                  "perspective": 850,
                                },
                                Object {
                                  "translateX": -100,
                                },
                                Object {
                                  "rotateY": "60deg",
                                },
                              ],
                            }
                          }
                        />
                  `);
    });
  });

  describe('when a style value is a hex colour', () => {
    const Component = () => <View style={{ color: '#c0ffee' }} />;

    const tree = renderer.create(<Component />).toJSON();

    it('should render the colour to the terminal', () => {
      expect(tree).toMatchInlineSnapshot(`
        <View
          style={
            Object {
              "color": "[7m[38;2;192;255;238m#c0ffee[39m[27m",
            }
          }
        />
      `);
    });
  });
});
