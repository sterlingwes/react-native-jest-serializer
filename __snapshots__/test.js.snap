// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`style prop serializing style with overrides and nested objects merge an array to the relevant style object 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": 10,
      "position": "absolute",
      "transform": Array [
        Object {
          "perspective": 850,
        },
        Object {
          "translateX": -100,
        },
        Object {
          "rotateY": "60deg",
        },
      ],
    }
  }
/>
`;

exports[`style prop serializing when a component has behaviour-related props should strip them from the snapshot output 1`] = `<View />`;

exports[`style prop serializing when a style value is a hex colour should render the colour to the terminal 1`] = `
<View
  style={
    Object {
      "color": "[7m[38;2;192;255;238m#c0ffee[39m[27m",
    }
  }
/>
`;

exports[`style prop serializing when there is no style prop should still render a snapsoht 1`] = `
<View
  coolView={true}
/>
`;
